synapse/handlers/message.py

```diff
@@ -901,6 +914,30 @@ def persist_and_notify_client_event(
                         "Room alias %s does not point to the room" % (room_alias_str,),
                     )
 
+            # Check that alt_aliases is the proper form.
+            alt_aliases = event.content.get("alt_aliases", [])
+            if not isinstance(alt_aliases, (list, tuple)):
+                raise SynapseError(400, "The alt_aliases property must be a list.")
```

Comment:
we should set an error code rather than using `M_UNKNOWN`. suggest `M_INVALID_PARAM` (see https://matrix.org/docs/spec/client_server/r0.6.0#api-standards)

(we should probably do the same for the exception raised by `RoomAlias.from_string` on invalid input)

Reply:
I made this change, although modifying `RoomAlias.from_string` seems to have fairly wide reaching affects that I'm unsure about!

Reply:
anything that relies on RoomAlias.from_string returning an errcode of `M_UNKNOWN` is pretty broken imho. I think we should be good here.